lib/tzinfo/timezone.rb

```diff
@@ -418,6 +418,21 @@ def period_for_local(local, dst = Timezone.default_dst)
       end
     end
 
+    def period_for(tm, dst = Timezone.default_dst)
```

Comment:
`period_for` doesn't need a `dst` parameter. There is no ambiguity to resolve since we know the UTC time (either directly or as a local time with an offset).